pkg/operator/controller/ingress/status.go

```diff
@@ -123,21 +131,143 @@ func computeIngressAvailableCondition(deployment *appsv1.Deployment) operatorv1.
 	}
 }
 
-func computeIngressDegradedCondition(deployment *appsv1.Deployment) operatorv1.OperatorCondition {
+// computeDeploymentDegradedCondition computes the ingresscontroller's
+// "DeploymentDegraded" status condition by examining the status conditions of
+// the deployment.
+func computeDeploymentDegradedCondition(deployment *appsv1.Deployment) operatorv1.OperatorCondition {
 	for _, cond := range deployment.Status.Conditions {
-		if cond.Type == appsv1.DeploymentProgressing && cond.Status == corev1.ConditionFalse && cond.Reason == "ProgressDeadlineExceeded" {
+		if cond.Type == appsv1.DeploymentAvailable && cond.Status == corev1.ConditionFalse && cond.Reason == "MinimumReplicasUnavailable" {
```

Comment:
Wouldn't we be degraded if the deployment were Available=False regardless of reason?

Reply:
That would make sense.  Should we directly reflect the deployment's status, reason, and message then?

Reply:
Seems like a good start... there's almost no limit to how far the analysis and aggregation can go... mirroring the reason would be way more info than we have now!

Reply:
Done: https://github.com/openshift/cluster-ingress-operator/compare/8b503220967687ec379a5e64d3a443172d2e6b70..a64df5779bcec0aeaf5c4f1e0c4e99a530c05da3